nixos/doc/manual/development/replace-modules.xml

```diff
@@ -0,0 +1,48 @@
+<section xmlns="http://docbook.org/ns/docbook"
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        xmlns:xi="http://www.w3.org/2001/XInclude"
+        version="5.0"
+        xml:id="sec-replace-modules">
+
+<title>Replace Modules</title>
+
+<para>It's also possible to disable modules, removing the option
+  declarations and config implementation defined within that file.
+  This can be used to import a set of modules from another channel
+  while keeping the rest of the system on a stable release.</para>
```

Comment:
nit: This is actually a really valid use case.  I suggest you add an additional 4 lines example which just does that assuming that `nixos-rebuild` is used with an extra `-I custom-nixos=/home/dev/nixos-custom/nixos` command line argument.